src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java

```diff
@@ -118,7 +110,7 @@ private void refusePostponement(SscsCaseData sscsCaseData) {
 
     private void grantPostponement(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response) {
         cancelHearing(sscsCaseData);
-        setHearingDateToExcludedDate(sscsCaseData, response);
+        postponementRequestService.addCurrentHearingToExcludeDates(response);
```

Comment:
This method is semantically different from setHearingDateToExcludedDate.

In particular, the selection of the hearing date in the previous one is done through `getHearingDateTime()` whereas in the new one it relies on `sscsCaseData.getLatestHearing` which aside from having side-effects is not ordering by `hearingDateTime` but by `id` of the hearing, which cannot be guaranteed will always represent chronological order.

Reply:
Disregard the part where I say that it sorts by `id`. It's counter-intuitive but the `hearingDateTime` is used first and the `id` last. So the only remaining complaint is due to `getLatestHearing` having side-effects, but that should be dealt with in its corresponding project.

This code is fine. Resolving the conversation.